dlt/destinations/weaviate/weaviate.py

```diff
@@ -0,0 +1,279 @@
+from types import TracebackType
+from typing import ClassVar, Optional, Sequence, Type, Iterable
+import base64
+import binascii
+import zlib
+
+import weaviate
+
+from dlt.common import json, pendulum, logger
+from dlt.common.schema import Schema, TTableSchema, TSchemaTables
+from dlt.common.schema.typing import VERSION_TABLE_NAME, LOADS_TABLE_NAME
+from dlt.common.destination import DestinationCapabilitiesContext
+from dlt.common.destination.reference import (
+    NewLoadJob,
+    TLoadJobState,
+    LoadJob,
+    JobClientBase,
+)
+from dlt.common.storages import FileStorage
+from dlt.destinations.job_impl import EmptyLoadJob
+from dlt.destinations.job_client_impl import StorageSchemaInfo
+
+from dlt.destinations.weaviate import capabilities
+from dlt.destinations.weaviate.configuration import WeaviateClientConfiguration
+
+DLT_TABLE_PREFIX = "_dlt"
+
+
+# TODO: move to common
+def is_dlt_table(table_name: str) -> bool:
+    return table_name.startswith(DLT_TABLE_PREFIX)
+
+
+def snake_to_camel(snake_str: str) -> str:
+    return "".join(x.capitalize() for x in snake_str.split("_"))
```

Comment:
interesting. we could easily create a naming convention so you have camel case in schema (in normalize stage)